ResourceStatusSystem/Command/test/Test_RSS_Command_GOCDBStatusCommand.py

```diff
@@ -3,488 +3,292 @@
-#                                                                          },
-#                                                      '669 devel.edu.mk 1': {
-#                                                        'HOSTED_BY': 'MK-01-UKIM_II',
-#                                                        'DESCRIPTION': 'Problem with SE server',
-#                                                        'SEVERITY': 'OUTAGE',
-#                                                        'HOSTNAME': 'devel.edu.mk 1',
-#                                                        'GOCDB_PORTAL_URL': 'myURL',
-#                                                        'FORMATED_END_DATE': '2013-07-20 00:00',
-#                                                        'FORMATED_START_DATE': newStartDate3
-#                                                                          }
-#                                                       }
-#                                         }
-#
-#     self.moduleTested.GOCDBClient.return_value = mock_GOCDB
-#     command = self.testClass( args = { 'element' : 'Site', 'name' : 'Y' } )
-#     res = command.doMaster()
-#
-#     self.assertTrue(res['OK'])
-#     self.assertEqual( [ 'DT', 'EndDate' ], res[ 'Value' ].keys() )
-#     self.assertEqual( 'OUTAGE', res[ 'Value' ][ 'DT' ] )
-#     self.assertEqual( '2013-07-20 00:00', res[ 'Value' ][ 'EndDate' ] )
-#
-#     # Restore the module
-#     self.moduleTested.GOCDBClient = self.mock_GOCDBClient
-#     reload( self.moduleTested )
-
-################################################################################
-# EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF
+"""
+Setup
+"""
+gLogger.setLevel('DEBUG')
+
+# Mock external libraries / modules not interesting for the unit test
+seMock = mock.MagicMock()
+seMock.options = {'SEType': 'T0D1'}
+
+mock_GOCDBClient = mock.MagicMock()
+mock_RMClient = mock.MagicMock()
+mock_RMClient.addOrModifyDowntimeCache.return_value = S_OK()
+
+args = {'name': 'aName', 'element': 'Resource', 'elementType': 'StorageElement'}
+
+
+def test_instantiate():
+  """ tests that we can instantiate one object of the tested class
+  """
+
+  command = DowntimeCommand()
+  assert command.__class__.__name__ == 'DowntimeCommand'
+
+
+def test_init():
+  """ tests that the init method does what it should do
+  """
+
+  command = DowntimeCommand()
+  assert command.args == {'onlyCache': False}
+  assert command.apis == {}
+
+  command = DowntimeCommand(clients={'GOCDBClient': mock_GOCDBClient})
+  assert command.args == {'onlyCache': False}
+  assert command.apis == {'GOCDBClient': mock_GOCDBClient}
+
+  command = DowntimeCommand(args)
+  _args = dict(args)
+  _args.update({'onlyCache': False})
+  assert command.args == _args
+  assert command.apis == {}
+
+
+def test_doCache(mocker):
+  """ tests the doCache method
+  """
+  mocker.patch("DIRAC.ResourceStatusSystem.Command.DowntimeCommand.StorageElement", return_value=seMock)
```

Comment:
Just for info, these patch can be added as decorator if you feel like

Reply:
I know. But I personally find it a bit more readable as it is.